### ISSUE TITLE
Datagrids lack accessible names

### ISSUE THREAD

**Issue by WhiteLord** [vmware/clarity#5461]
## Describe the accessibility issue

A part of our app consists in a screen with several datagrids that have a title above them, 
After a11y audit, it is stated that `Assistive technologies will not be able to programmatically identify these grids and convey their purpose to users.`
I'd like to create a discussion whether this is actually needed, as I'm wondering what the accessibility review for the current implementation of the datagrid is.

## WCAG Guideline

Success Criterion 4.1.2 Name, Role, Value
(Level A)
https://www.w3.org/TR/WCAG21/#name-role-value

## Steps to reproduce the behavior:

1. Create a basic datagrid as the one provided in the examples
2. Observe that there is no possibility to edit or provide any aria-label or aria-labelledby

## Expected behavior

Provide a possibility to include an aria-label or aria-labelledby to datagrids

## Versions

**App**

Currently no version supports this. If included, I think it will be best to backport it.

**Device:**

- Type: [MBP]
- OS: [iOS]
- Browser [FF, Chrome]
- Version [TOT]
- Screen Reader: [NVDA]

For screen readers, we primarily support the following combinations, and other combinations will be considered on a case by case basis.

- NVDA + Firefox (Windows)
- VoiceOver + Safari (Apple)

## Additional notes

_Add any other notes about the problem here._

**Comment by Jinnie** [comment]
@chlane I am not sure I understand what is required here, can you please take a look if it is valid and if we already track it internally.

**Comment by WhiteLord** [comment]
I'm guessing that, according to the a11y review we got, we should have the possibility to add `aria-label` or `aria-labelledby` to the datagrid. 
I come to this conclusion from the following lines in our review:

```
See the relevant HTML/CSS below.

<div class=""datagrid-table"" role=""grid"">

[User Impact]
Assistive technologies will not be able to programmatically identify these grids and convey their purpose to users. 
```

**Comment by chlane** [comment]
This issue in now tracked internally by our accessibility team

**Comment by WhiteLord** [comment]
Thanks for the update. Please, do close it if it's tracked elsewhere so that we avoid duplication.

**Comment by chlane** [comment]
I think someone on the Clarity team needs to close it I don't see the option
